tests/zfs-tests/tests/functional/rsend/rsend_025_pos.ksh

```diff
@@ -0,0 +1,93 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+
+#
+# Copyright (c) 2013, 2016 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/rsend/rsend.kshlib
+
+# DESCRIPTION:
+#	zfs send -exclude will exclude the given hierarchy
+#	and only that given hierarchy.
+#
+# STRATEGY:
+#	1. Setup test model
+#	2. Create several datasets on pool.
+#	3. Send -R -X pool/dataset
+#	4. Verify receive does not have the excluded dataset(s).
+
+verify_runnable "both"
+
+function cleanup
+{
+    cleanup_pool $POOL2
+    cleanup_pool $POOL
+    log_must setup_test_model $POOL
+}
+
+log_assert "zfs send -R -X will skip excluded dataset(s)"
+log_onexit cleanup
+
+cleanup
+
+#
+# Create some datasets
+for ds in ds1 ds1/second ds1/second/third \
+	      ds2 ds2/second
+do
+    log_must zfs create $POOL/$ds
+done
+
+log_must zfs create snapshot -r $POOL@presend
+
+log_must eval "zfs send -R $POOL@presend > $BACKDIR/presend"
+log_must eval "zfs receive -d -F $POOL2 < $BACKDIR/presend"
+
+for ds in ds1 ds1/second ds1/second/third \
+	      ds2 ds2/second
+do
+    log_must datasetexists $POOL2/$ds
+done
+
+log_must_busy zfs destroy -rf $POOL2@presend
+
+log_must eval "zfs send -R -X ds1/second > $BACKDIR/presend"
```

Comment:
We need more (separate!) tests that assert multiple -X works.

Reply:
Well, I wrote the one because it was requested, but because I couldn't get it to work locally I didn't write more.  Since it *does* seem to have worked (right?), that gives me a template for adding a couple more...

Reply:
Ok.  The tests all fail for me.  However, it looks like they fail for everyone (based on a couple of open pull requests I just checked), so I don't think it's me.  That said:  I don't know if the test I created actually works!  So I'm not sure what to do here.

Reply:
Well, there is a continuous wrestling match between tests being flaky and whack a mole, but every test bot failing is generally still a sign of a problem.

If you [look](http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20%28TEST%29/builds/19261/steps/shell_4/logs/tests) at e.g. the Centos 7 bot, the only test that failed is the one you added, and if you [look](http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20%28TEST%29/builds/19261/steps/shell_4/logs/log) at the detailed output:

```
Test: /usr/share/zfs/zfs-tests/tests/functional/rsend/rsend_025_pos (run as root) [00:03] [FAIL]
22:06:58.12 ASSERTION: zfs send -R -X will skip excluded dataset(s)
22:06:58.12 SUCCESS: rm -rf /mnt/backdir-rsend/*
22:06:58.14 SUCCESS: zfs destroy -Rf testpool2
22:06:58.15 SUCCESS: rm -rf /mnt/backdir-rsend/*
22:06:58.48 SUCCESS: zfs destroy -Rf testpool
22:06:58.56 SUCCESS: zfs create -p testpool/testfs/fs1/fs2
22:06:58.58 SUCCESS: zfs snapshot testpool@psnap
22:06:58.60 SUCCESS: zfs clone testpool@psnap testpool/pclone
22:06:58.64 SUCCESS: zfs create -V 16M testpool/vol
22:06:58.66 SUCCESS: zfs create -V 16M testpool/testfs/vol
22:06:58.85 SUCCESS: zfs snapshot testpool/testfs/vol@vsnap
22:06:58.88 SUCCESS: zfs clone testpool/testfs/vol@vsnap testpool/testfs/vclone
22:06:59.08 SUCCESS: snapshot_tree testpool/testfs/fs1/fs2@fsnap
22:06:59.10 SUCCESS: zfs clone testpool/testfs/fs1/fs2@fsnap testpool/testfs/fs1/fclone
22:06:59.13 SUCCESS: zfs snapshot -r testpool@init
22:06:59.17 SUCCESS: snapshot_tree testpool@snapA
22:06:59.20 SUCCESS: snapshot_tree testpool@snapC
22:06:59.31 SUCCESS: snapshot_tree testpool/pclone@snapB
22:06:59.33 SUCCESS: snapshot_tree testpool/testfs@snapB
22:06:59.41 SUCCESS: snapshot_tree testpool/testfs@snapC
22:06:59.44 SUCCESS: snapshot_tree testpool/testfs/fs1@snapA
22:06:59.51 SUCCESS: snapshot_tree testpool/testfs/fs1@snapB
22:06:59.55 SUCCESS: snapshot_tree testpool/testfs/fs1@snapC
22:06:59.58 SUCCESS: snapshot_tree testpool/testfs/fs1/fclone@snapA
22:06:59.60 SUCCESS: zfs snapshot testpool/vol@snapA
22:06:59.61 SUCCESS: zfs snapshot testpool/testfs/vol@snapB
22:06:59.63 SUCCESS: zfs snapshot testpool/testfs/vol@snapC
22:06:59.64 SUCCESS: zfs snapshot testpool/testfs/vclone@snapC
22:06:59.67 SUCCESS: zfs snapshot -r testpool@final
22:06:59.68 SUCCESS: setup_test_model testpool
22:06:59.70 SUCCESS: zfs create testpool/ds1
22:06:59.73 SUCCESS: zfs create testpool/ds1/second
22:06:59.76 SUCCESS: zfs create testpool/ds1/second/third
22:06:59.78 SUCCESS: zfs create testpool/ds2
22:06:59.81 SUCCESS: zfs create testpool/ds2/second
22:06:59.81 invalid option 'r' usage: create [-Pnpuv] [-o property=value] ... <filesystem> create [-Pnpsv] [-b blocksize] [-o property=value] ... -V <size> <volume> For the property list, run: zfs set|get For the delegated permission list, run: zfs allow|unallow
22:06:59.81 ERROR: zfs create snapshot -r testpool@presend exited 2
```